mpower/src/main/java/org/sagebionetworks/research/mpower/show_step_fragment/ShowOverviewStepFragment.java

```diff
@@ -76,6 +98,29 @@ public static ShowOverviewStepFragment newInstance(@NonNull StepView stepView) {
         return fragment;
     }
 
+    @Override
+    public void onStart() {
+        super.onStart();
+        DisablableScrollView scrollView = this.stepViewBinding.getScrollView();
+        if (this.isFirstRun) {
```

Comment:
I'm thinking first run will end up being recorded in SharedPreferences. Probably a timestamp with the taskIdentifier as the key. Unless we need to record the taskResult from first run to get other info?

Reply:
So this one gets a little complicated. There are kind of 2 separate issues here,
1. Currently the last run is recorded in SharedPreferences at the end of the task run, by the showStep() method in PerformTaskFragment. It is then read in the call to getLastRun() from PerformTaskFragment's onCreate(), and passed to the PerformTaskViewModel which stores it as a ZonedDateTime.
2. This value is then written to the TaskResult which isn't strictly necessary. My thinking on this one was that this is how the iOS side works, and that the researchers might ultimately want this information so they know what was displayed to the user.

Looking back through my code I also stored this value in PerformTaskViewModel which is totally unnecessary and I will refactor that out.